src/gui/newwizard/SetupWizardWindow.cpp

```diff
@@ -0,0 +1,152 @@
+
+    connect(_ui->cancelButton, &QPushButton::clicked, this, &SetupWizardWindow::reject);
+
+    // set up transition
+    //    _transitionLayout->addWidget(_transitionProgressIndicator.get());
+
+    connect(_ui->nextButton, &QPushButton::clicked, this, [this]() {
+        slotStartTransition();
+        emit nextButtonClicked(_pagination->activePageIndex());
+    });
+
+    connect(_ui->backButton, &QPushButton::clicked, this, [this]() {
+        slotStartTransition();
+        emit backButtonClicked(_pagination->activePageIndex());
+    });
+
+    connect(_pagination.get(), &Pagination::paginationEntryClicked, this, [this](PageIndex clickedPageIndex) {
+        slotStartTransition();
+        emit paginationEntryClicked(_pagination->activePageIndex(), clickedPageIndex);
+    });
+
+    _transitionProgressIndicator->setFixedSize(32, 32);
+}
+
+void SetupWizardWindow::displayPage(AbstractSetupWizardPage *page, PageIndex index)
+{
+    // save some CPU cycles
+    _transitionProgressIndicator->stopAnimation();
+
+    _ui->backButton->setEnabled(true);
+    _ui->nextButton->setEnabled(true);
+
+    if (index == 0) {
+        _ui->backButton->setEnabled(false);
+    } else if (index == _pagination->size()) {
+        _ui->nextButton->setEnabled(false);
+    }
+
+    if (index >= (_pagination->size() - 1)) {
+        _ui->nextButton->setText(tr("Finish"));
+    } else {
+        _ui->nextButton->setText(tr("Next >"));
+    }
+
+    _currentPage = page;
+    slotReplaceContent(_currentPage);
+    // handle user pressing enter/return key
+    _currentPage->installEventFilter(this);
+
+    _pagination->setActivePageIndex(index);
+    _pagination->setEnabled(true);
+
+    connect(_ui->errorMessageDismissButton, &QPushButton::clicked, this, &SetupWizardWindow::slotHideErrorMessageWidget);
+
+    // by default, set focus on the next button
+    _ui->nextButton->setFocus();
+
+    // this can optionally be overwritten by the page
+    _currentPage->slotOnPageDisplayed();
+}
+
+void SetupWizardWindow::slotStartTransition()
+{
+    _transitionProgressIndicator->startAnimation();
+    slotReplaceContent(_transitionProgressIndicator.get());
+
+    // until a new page is displayed by the controller, we want to prevent the user from initiating another page change
+    _ui->backButton->setEnabled(false);
+    _ui->nextButton->setEnabled(false);
+    _pagination->setEnabled(false);
+    // also, we should assume the user has seen the error message in case one is shown
+    slotHideErrorMessageWidget();
+}
+
+void SetupWizardWindow::slotReplaceContent(QWidget *newWidget)
+{
+    // there should not be more than one widget
+    for (decltype(_contentLayout->count()) i = 0; i < _contentLayout->count(); ++i) {
+        _contentLayout->removeItem(_contentLayout->itemAt(i));
+    }
+
+    _contentLayout->addWidget(newWidget);
+}
+
+void SetupWizardWindow::slotHideErrorMessageWidget()
+{
+    _ui->errorMessageWidget->setMaximumSize(16777215, 0);
```

Comment:
???

Reply:
The first value is something I picked from Qt creator, some integer maximum of some kind, I'd say.